### PR TITLE
ref(replay): Refactor Replay Timeline Events to use *Frame types

### PR DESCRIPTION
**Pull request by ryan953**

### CODE REVIEW

File: static/app/components/replays/breadcrumbs/breadcrumbItem.tsx [hunk 1]
```diff
@@ -64,7 +92,8 @@ function BreadcrumbItem({
   startTimestampMs,
   style,
 }: Props) {
-  const {color, description, projectSlug, title, type} = getDetails(crumb);
+  const {color, description, projectSlug, title, type, timestampMs} =
+    getCrumbOrFrameData(crumb);
```
Comment (ryan953) [inline static/app/components/replays/breadcrumbs/breadcrumbItem.tsx:96]: this `getCrumbOrFrameData` doesn't need to stick around forever.

But when i'm using `frame.tsx` it's kind of annoying to have 4 different functions (8 switch statements) to keep in sync, so i might refactor that to be one method that returns everything (color, title, desc, icon) as a single object. I think only `getColor` is called individually, all others are needed at the same time.

File: static/app/utils/replays/replayReader.tsx [hunk 2]
```diff
@@ -252,17 +251,18 @@ export default class ReplayReader {
...
   getChapterFrames = memoize(() =>
     [
-      ...this._sortedBreadcrumbFrames.filter(
-        frame =>
-          [
-            'replay.init',
-            'ui.click',
-            'replay.mutations',
-            'ui.slowClickDetected',
-          ].includes(frame.category) || !BreadcrumbCategories.includes(frame.category)
```
Comment (ryan953) [inline static/app/utils/replays/replayReader.tsx:262]: including `!BreadcrumbCategories.includes(frame.category)` means that all the custom crumbs are included.

These should be part of the console instead.

File: static/app/components/replays/breadcrumbs/breadcrumbItem.tsx [hunk 3]
```diff
@@ -131,7 +159,7 @@ function BreadcrumbItem({
             />
           </InspectorWrapper>
         )}
-        {shouldShowCrumbProject && <CrumbProject projectSlug={projectSlug} />}
+        {projectSlug ? <CrumbProject projectSlug={projectSlug} /> : null}
```
Comment (billyvg) [inline static/app/components/replays/breadcrumbs/breadcrumbItem.tsx:162]: Will this show project when `crumb.type !== BreadcrumbType.ERROR`?

File: static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx [hunk 4]
```diff
@@ -101,15 +98,15 @@ function Event({
   `;
 
   // If we have more than 3 events we want to make sure of showing all the different colors that we have
-  const uniqueColors = Array.from(new Set(crumbs.map(crumb => crumb.color)));
+  const uniqueColors = uniq(frames.map(frame => getColor(frame)));
```
Comment (billyvg) [inline static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx:101]: Could simplify to 

```suggestion
  const uniqueColors = uniq(frames.map(getColor));
```

File: static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx [hunk 5]
```diff
@@ -121,29 +118,28 @@ const IconNodeTooltip = styled(Tooltip)`
   align-items: center;
 `;
 
-const IconPosition = styled('div')<{markerWidth: number}>`
+const IconPosition = styled('div')`
```
Comment (billyvg) [inline static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx:121]: 👍

File: static/app/utils/replays/replayReader.tsx [hunk 6]
```diff
@@ -259,10 +258,12 @@ export default class ReplayReader {
             'ui.click',
             'replay.mutations',
             'ui.slowClickDetected',
-          ].includes(frame.category) || !BreadcrumbCategories.includes(frame.category)
+            'navigation',
+          ].includes(frame.category) ||
+          (frame.category === 'ui.multiClick' && frame.data.clickCount >= 3)
       ),
       ...this._sortedSpanFrames.filter(frame =>
-        ['navigation.navigate', 'navigation.reload', 'largest-contentful-paint'].includes(
```
Comment (billyvg) [inline static/app/utils/replays/replayReader.tsx:265]: What happens to LCP?

File: static/app/utils/replays/types.tsx [hunk 7]
```diff
@@ -15,7 +15,21 @@ import type {
 } from '@sentry/replay';
 import invariant from 'invariant';
 
-export type RawBreadcrumbFrame = TRawBreadcrumbFrame;
+/**
+ * Extra types to account for stock SDK breadcrumbs
+ */
+type NonSDKBreadcrumbs = {
```
Comment (billyvg) [inline static/app/utils/replays/types.tsx:21]: This name seems to conflict with the comment re: "stock SDK breadcrumbs"

**Review by billyvg — APPROVED** [review]

**Pull request merged**